### PR TITLE
Add WAN eviction cluster properties [HZ-2619]

### PR DESCRIPTION
**Pull request by JamesHazelcast**
We recently added 2 new `ClusterProperty` entries for 5.4 related to the replication of eviction events with `IMap` and `ICache`.

Relevant PR: https://github.com/hazelcast/hazelcast/pull/24941

### CODE REVIEW

**Comment by JamesHazelcast** [comment]
This will also need backporting for the next `5.3.z` release as we are backporting this enhancement.

File: docs/modules/ROOT/pages/system-properties.adoc [hunk 1]
```diff
@@ -1164,6 +1164,16 @@ master starts forming the cluster.
 |Defines the pending invocation threshold for the WAN replication implementation. Exceeding this threshold on a WAN consumer member makes the member to delay the WAN acknowledgment,
 thus slowing down the WAN publishers on the source side that send WAN events to the given WAN consumer. Setting this value to negative disables the acknowledgement delaying feature.
 
+|`hazelcast.wan.replicate.imap.evictions`
+|false
+|bool
+|Specifies whether `IMap` objects should replicate entry evictions over WAN to target clusters. When enabled, ALL evictions (size- or expiration-driven as well as manual invocation) are replicated to targets, so this should be used with caution. Setting this property to true does not guarantee data consistency between two clusters.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/system-properties.adoc:1170]:
```suggestion
|Specifies whether `IMap` objects replicate entry evictions over WAN to target clusters. Use this property with caution. When enabled, _ALL_ evictions, including size or expiration-driven evictions and manual invocations are replicated to target clusters. Setting this property to `true` does not guarantee data consistency between clusters.
```

File: docs/modules/ROOT/pages/system-properties.adoc [hunk 2]
```diff
@@ -1164,6 +1164,16 @@ master starts forming the cluster.
...
+
+|`hazelcast.wan.replicate.icache.evictions`
+|false
+|bool
+|Specifies whether `ICache` objects should replicate entry evictions over WAN to target clusters. When enabled, ALL evictions (size- or expiration-driven as well as manual invocation) are replicated to targets, so this should be used with caution. Setting this property to true does not guarantee data consistency between two clusters.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/system-properties.adoc:1175]:
```suggestion
|Specifies whether `ICache` objects replicate entry evictions over WAN to target clusters. Use this property with caution. When enabled, _ALL_ evictions, including size or expiration-driven evictions and manual invocations are replicated to target clusters. Setting this property to `true` does not guarantee data consistency between clusters.
```

**Review by angelasimms — CHANGES_REQUESTED** [review]
Minor changes for clarity. Please let me know if there are any questions.

**Comment by JamesHazelcast** [comment]
> Minor changes for clarity. Please let me know if there are any questions.

Appreciate the feedback @angelasimms, definitely much more clear now, thank you 👍

**Review by angelasimms — APPROVED** [review]

**Comment by Serdaro** [comment]
@JamesHazelcast , @angelasimms, thank you for the content and review.
I believe this topic deserves a bit deeper insight in the Synchronizing Data Across Clusters section of the Platform documentation, since the System Properties section is mostly a reference.

If it is a fine-tuning feature (which I think it is, since the defaults are `false`), you can open up a "Replicating Map/Cache Evictions" or simply "Replicating Evictions" subsection under the Fine-Tuning WAN Replication section.

The content for this new section can talk about why the user may need to enable eviction replications (with some basic scenarios), in what ways it can affect the data consistency when enabled and in use, which points to be taken into consideration in terms of "use with caution", and finally an example WAN replication configuration containing these new properties.

**Comment by JamesHazelcast** [comment]
Great suggestion @Serdaro, thank you.

I've made a commit which adds more in-depth documentation as you suggested. Please let me know if you'd like any further changes made 👍

File: docs/modules/wan/pages/tuning.adoc [hunk 3]
```diff
@@ -754,3 +754,47 @@ hazelcast:
...
+The replication of `IMap` and `ICache` evictions to other clusters can be enabled by setting the relevant xref:ROOT:system-properties.adoc[Hazelcast
+system property] to `true`. The replication of eviction events can be useful for maintaining stronger data consistency
+between clusters, however it should be used with caution: source clusters with eviction enabled should replicate to
+target clusters which **do not** have eviction enabled. If both source and target clusters have eviction enabled, due
+to eviction being a largely local operation, it is possible for a target cluster to evict data that has not been
+evicted on the source cluster. When replication is used for data consistency on a failover deployment, it is important
+that eviction is re-enabled on the standby cluster when it becomes active.
+
+While enabling this functionality can help to maintain stronger data consistency over WAN between clusters, it is
+important to note that this **does not guarantee full data consistency**.
```
Comment (Serdaro) [inline docs/modules/wan/pages/tuning.adoc:769]: Awesome content @JamesHazelcast , thanks a lot.
I'd suggest to move the very first sentence of this section to here (right before the configuration examples), with a minor editing:

"You can enable the replication of `IMap` and `ICache` evictions to the other clusters by setting the `hazelcast.wan.replicate.imap/icache.evictions` property to `true`, as shown below."

File: docs/modules/wan/pages/tuning.adoc [hunk 4]
```diff
@@ -754,3 +754,47 @@ hazelcast:
         use-endpoint-private-address: true
 ----
 ====
+
+== Replicating `IMap`/`ICache` Evictions
+
+The replication of `IMap` and `ICache` evictions to other clusters can be enabled by setting the relevant xref:ROOT:system-properties.adoc[Hazelcast
+system property] to `true`. The replication of eviction events can be useful for maintaining stronger data consistency
+between clusters, however it should be used with caution: source clusters with eviction enabled should replicate to
```
Comment (Serdaro) [inline docs/modules/wan/pages/tuning.adoc:762]:
```suggestion
between clusters. However, you should use it with caution; source clusters with eviction enabled should replicate to
```

File: docs/modules/wan/pages/tuning.adoc [hunk 5]
```diff
@@ -754,3 +754,47 @@ hazelcast:
...
+[source,yaml]
+----
+hazelcast:
+    ...
+    properties:
+      hazelcast.wan.replicate.imap.evictions: true
+      hazelcast.wan.replicate.icache.evictions: true
+    ...
+----
+====
```
Comment (Serdaro) [inline docs/modules/wan/pages/tuning.adoc:800]: And, after the configuration examples, we can give the reference to the System Properties content:

"You can find the specifications of these properties in xref:ROOT:system-properties.adoc[System Properties]."

**Review by Serdaro — COMMENTED** [review]
Minor comments.

**Review by Serdaro — APPROVED** [review]

**Comment by Serdaro** [comment]
Let's merge this just before Platform 5.3.2 is released.

**Pull request merged**